src/components/MultiHopTrade/components/TradeQuotes/TradeQuotes.tsx

```diff
@@ -1,24 +1,33 @@
 import { Collapse, Flex } from '@chakra-ui/react'
-import { useGetTradeQuotes } from 'components/MultiHopTrade/hooks/useGetTradeQuotes'
+import { selectQuotes, selectSelectedSwapperName } from 'state/slices/tradeQuoteSlice/selectors'
+import { useAppSelector } from 'state/store'
 
+import type { TradeQuoteLoadedProps } from './TradeQuote'
 import { TradeQuote } from './TradeQuote'
 
 type TradeQuotesProps = {
   isOpen?: boolean
 }
 
 export const TradeQuotes: React.FC<TradeQuotesProps> = ({ isOpen }) => {
-  const { selectedQuote, sortedQuotes } = useGetTradeQuotes()
+  const sortedQuotes = useAppSelector(selectQuotes)
+  const activeSwapperName = useAppSelector(selectSelectedSwapperName)
 
-  const activeSwapperName = selectedQuote?.swapperName
-  const bestQuoteData = sortedQuotes[0]
+  const bestQuoteData = sortedQuotes?.[0]
 
-  const quotes = sortedQuotes.map((quoteData, i) => {
-    const { data, swapperName } = quoteData
-    const quote = data?.isOk() ? data.unwrap() : undefined
+  const quotes = sortedQuotes?.map((apiQuote, i) => {
+    const { quote, swapperName, inputOutputRatio, error } = apiQuote
+
+    const quoteDataShim: TradeQuoteLoadedProps['quoteData'] = {
+      isLoading: false,
```

Comment:
loading state will need to be plumbed through from the api

Reply:
you could potentially store the loading state in the store, not sure if that breaks our idea of 1 source of truth though